tests/DependencyInjection/Compiler/AddProvidersPassTest.php

```diff
@@ -84,4 +87,50 @@ public function testProcessForManyProviders()
         $providersPass = new AddProvidersPass();
         $providersPass->process($containerBuilderMock);
     }
+
+    public function testPriorityRegisteredProviders()
+    {
+        $expectedProviders = [];
+        $expectedProviders[0][] = new Reference('id');
+        $expectedProviders[20][] = new Reference('id2');
+
+        krsort($expectedProviders);
+        $expectedProviders = call_user_func_array('array_merge', $expectedProviders);
+
+        if (class_exists(IteratorArgument::class)) {
+            $expectedProviders = new IteratorArgument($expectedProviders);
+        }
+
+        $definitionMock = $this->getMockBuilder('Symfony\Component\DependencyInjection\Definition')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $definitionMock->expects($this->once())
+            ->method('replaceArgument')
+            ->with($this->equalTo(0), $expectedProviders);
+
+        $containerBuilderMock = $this->getMockBuilder('Symfony\Component\DependencyInjection\ContainerBuilder')->getMock();
+        $containerBuilderMock->expects($this->once())
+            ->method('hasDefinition')
+            ->will($this->returnValue(true));
+        $containerBuilderMock->expects($this->once())
+            ->method('findTaggedServiceIds')
+            ->with($this->equalTo('knp_menu.provider'))
+            ->will($this->returnValue([
+                'id2' => ['provider_tag2', ['priority' => 20]],
```

Comment:
can you add the example with multiple tags here, to cover that edge case too? and use -20 for this provider, so the krsort actually does reverse something?